src/main/java/com/napier/sem/DatabaseConnectionImpl.java

```diff
@@ -0,0 +1,133 @@
+package com.napier.sem;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DatabaseConnectionImpl implements DatabaseConnection {
+
+    private static DatabaseConnectionImpl instance;
+    static final int RETRIES = 10;
+    /**
+     * Connection to MySQL database.
+     */
+    private Connection connection = null;
+
+    private DatabaseConnectionImpl(){
+    }
+
+    public static DatabaseConnection getInstance(){
+        if (instance == null){
+            instance = new DatabaseConnectionImpl();
+        }
+        return instance;
+    }
+
+    /**
+     * Accepts an SQL statement, executes it and returns the result of the query
+     * @param request a string in the form of an SQL statement
+     * @return List of results from the database. Returns null if there was an error.
+     */
+    @Override
+    public List<Map<String, Object>> executeSQLStatement(String request) {
```

Comment:
Why do we return a List<Map<String, Object>>? Can we not just return a string? I think all we have to do with the output of a query is print it